src/include/miopen/tensor_ops.hpp

```diff
@@ -75,7 +75,13 @@ GetConsistentFlattenedTensorDescriptors(const TDescriptors&... real_descriptor_p
     for(std::size_t itensor = 0; itensor < NTensor; ++itensor)
         is_all_packed &= real_descriptors[itensor]->IsPacked();
 
-    if(is_all_packed)
+    bool is_all_same_strided        = true;
+    const auto& real_desc_0_strides = real_descriptors[0]->GetStrides();
+    for(std::size_t itensor = 0; itensor < NTensor; ++itensor)
+        if(real_desc_0_strides != real_descriptors[itensor]->GetStrides())
+            is_all_same_strided = false;
```

Comment:
1) Strides with index 0 has been compared to itself
2) Early exit

```suggestion
    const auto& real_desc_0_strides = real_descriptors[0]->GetStrides();
    for(std::size_t itensor = 1; itensor < NTensor; ++itensor)
        if(real_desc_0_strides != real_descriptors[itensor]->GetStrides())
        {
            is_all_same_strided = false;
            break;
        }
```